plugins/rest/aws.go

```diff
@@ -211,7 +212,12 @@ func (cs *awsMetadataCredentialService) urlForMetadataService() (string, error)
 	// otherwise, check environment to see if it looks like we're in an ECS
 	// container (with implied role association)
 	if isECS() {
-		return ecsDefaultCredServicePath + os.Getenv(ecsRelativePathEnvVar), nil
+		// first check if the relative env var exists; if so we use that otherwise we
+		// use the "full" var
+		if _, relativeExists := os.LookupEnv(ecsRelativePathEnvVar); relativeExists {
+			return ecsDefaultCredServicePath + os.Getenv(ecsRelativePathEnvVar), nil
+		}
+		return os.Getenv(ecsFullPathEnvVar), nil
```

Comment:
What happens if `AWS_CONTAINER_CREDENTIALS_FULL_URI ` is not set? We probably need to fall back on `ecsDefaultCredServicePath + os.Getenv(ecsRelativePathEnvVar)`. Otherwise won't this return an empty string?

Reply:
I might be misunderstanding your commend, but line 218 checks for the existence of `ecsRelativePathEnvVar` and uses that if it exists. If it doesn't then it falls back to the new variable

Reply:
One must exist because `isECS()` is true

Reply:
My bad, I missed the change you made to the `isECS()` method. Any reason to keep the PR in draft mode?

Reply:
> Any reason to keep the PR in draft mode?

This doesn't actually work at the moment I don't think - Im building and testing a version my end but when I've confirmed it working I'll push the fixes and mark ready for review.

Thanks for your help so far 👍